src/xai_grok/grok.py

```diff
@@ -0,0 +1,223 @@
+            http.HTTPMethod.POST,
+            endpoint,
+            request_body=request.model_dump_json(exclude_none=True),
+        )
+        if response.status_code == http.client.BAD_REQUEST:
+            raise errors.InvalidRequestError(response.text)
+        elif response.status_code == http.client.UNPROCESSABLE_ENTITY:
+            raise errors.IncompleteRequestError(response.text)
+        return self._parse_response_as_type(
+            response, schemas.CompleteResponse, endpoint=endpoint
+        )
+
+    def completions(self, request: schemas.SampleRequest) -> schemas.SampleResponse:
+        endpoint = "v1/completions"
+        self.logger.warning(
+            "Using legacy endpoint: POST https://api.x.ai/v1/completions. "
+            "This endpoint is deprecated, please use POST https://api.x.ai/v1/chat/completions instead."
+        )
+        response = self._send_request(
+            http.HTTPMethod.POST,
+            endpoint,
+            request_body=request.model_dump(exclude_none=True),
+        )
+        if response.status_code == http.client.BAD_REQUEST:
+            raise errors.InvalidRequestError(response.text)
+        if response.status_code == http.client.UNPROCESSABLE_ENTITY:
+            raise errors.IncompleteRequestError(response.text)
+        return self._parse_response_as_type(
+            response, schemas.SampleResponse, endpoint=endpoint
+        )
+
+    def embedding_model(self, model_id: str) -> schemas.EmbeddingModel:
+        endpoint = f"v1/embedding-model/{model_id}"
+        response = self._send_request(http.HTTPMethod.GET, endpoint)
+        if response.status_code == http.client.NOT_FOUND:
+            raise errors.ModelNotFoundError(f"Embedding Model {model_id} not found")
+        return self._parse_response_as_type(
+            response, schemas.EmbeddingModel, endpoint=endpoint
+        )
+
+    def embedding_models(self) -> schemas.EmbeddingModels:
+        endpoint = "v1/embedding-models"
+        response = self._send_request(http.HTTPMethod.GET, endpoint)
+        return self._parse_response_as_type(
+            response, schemas.EmbeddingModels, endpoint=endpoint
+        )
+
+    def embeddings(
+        self, request: schemas.EmbeddingRequest
+    ) -> schemas.EmbeddingResponse:
+        endpoint = "v1/embeddings"
+        response = self._send_request(
+            http.HTTPMethod.POST,
+            endpoint,
+            request_body=request.model_dump_json(exclude_none=True),
+        )
+        if response.status_code == http.client.BAD_REQUEST:
+            raise errors.InvalidRequestError(response.text)
+        if response.status_code == http.client.UNPROCESSABLE_ENTITY:
+            raise errors.IncompleteRequestError(response.text)
+        return self._parse_response_as_type(
+            response, schemas.EmbeddingResponse, endpoint=endpoint
+        )
+
+    def language_model(self, model_id: str) -> schemas.LanguageModel:
```

Comment:
Typo in the endpoint URL for `language_model` method: 'langauge-models' instead of 'language-models', which will cause API requests to fail.